### ISSUE TITLE
Would you consider releasing the SFT stage code?

### ISSUE THREAD

**Issue by CeleBin** [Lanyu0303/KPO#1]
Hello, thank you very much for sharing this project!

I noticed that the repository currently provides the code for KPO, but the Supervised Fine-Tuning (SFT) stage code has not been released yet.
Would it be possible to make the SFT stage code public as well?

This would greatly help researchers to better understand the complete training pipeline and further reproduce or extend the work.
Thank you for your consideration!

**Comment by Lanyu0303** [comment]
Sorry for the late reply. We are using the Swift framework directly (https://github.com/modelscope/ms-swift) for SFT. You can refer to their official documentation, or check out the following script:

NPROC_PER_NODE=2  MASTER_PORT=29520 \
CUDA_VISIBLE_DEVICES=0,1 swift sft \
    --model_id_or_path $MODEL_PATH \
    --model_type 'llama3_2-3b-instruct'\
    --model_revision master \
    --sft_type lora \
    --tuner_backend peft \
    --template_type llama3_2 \
    --dtype AUTO \
    --output_dir $OUTPUT_DIR \
    --dataset $DATA_PATH \
    --val_dataset $VAL_DATA_PATH \
    --train_dataset_sample -1 \
    --num_train_epochs 5 \
    --max_length 2048 \
    --check_dataset_strategy warning \
    --lora_rank 32 \
    --lora_alpha 64 \
    --lora_dropout_p 0.1 \
    --lora_target_modules ALL \
    --gradient_checkpointing true \
    --batch_size 32 \
    --weight_decay 0.1 \
    --learning_rate 1e-4 \
    --gradient_accumulation_steps 2 \
    --max_grad_norm 0.5 \
    --warmup_ratio 0.03 \
    --eval_steps 300 \
    --save_steps 300 \
    --save_total_limit 30 \
    --logging_steps 1

**Comment by CeleBin** [comment]
Thank you very much for your detailed reply and for pointing me to the Swift framework for the SFT stage — that was very helpful!

I still have one remaining question regarding the dataset. In the file dataset/movielens_dataset.py, it seems that the following files are required:

data/ref/movielens/u.item
data/ref/movielens/train_data.df
data/ref/movielens/val_data.df
data/ref/movielens/test_data.df

I was able to locate the u.item file (for example from the [MovieLens-100k dataset on HuggingFace](https://huggingface.co/datasets/includeno/movielens-100k/tree/main)
), but I could not find the other three files (train_data.df, val_data.df, test_data.df).

Could you please clarify how these .df files were generated (e.g., from the raw MovieLens data)? Or if possible, would it be possible to provide them directly?

Thanks again for your time and for sharing this project!

**Comment by Lanyu0303** [comment]
Sorry for the late reply. The data has been uploaded to the GitHub repository. Thank you for your patience.

**Comment by CeleBin** [comment]
I sincerely apologize for bothering you again, and thank you for your response. While reproducing the process, I noticed that the process_data() function requires not only correct_answer and select_k, but also original_prompt and rejectList. However, these two fields do not seem to be included in the processing logic of movielens_dataset.py. May I confirm if my understanding is correct that the seq_str produced by movielens_dataset.py corresponds to original_prompt, and cans_name corresponds to rejectList?

**Comment by Lanyu0303** [comment]
1. The `original_prompt` corresponds to `seq_str`, and it can be constructed by filling in a template similar to the following:

```python
original_prompt = '''This person has previously watched {}. Please predict the next movie the person will watch. Choose the answer from the following 20 movie titles: {}. Answer: '''.format(seq_str, candidate_items)
```

2. The rejectList includes candidate items except for the correct item, sorted in descending order based on their logits.
